src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.Patch.cs

```diff
@@ -185,6 +185,10 @@ public abstract partial class InMemoryDocumentSessionOperations
             var pathScript = path.CompileToJavascript(_javascriptCompilationOptions.Value);
 
             var valueToUse = AddTypeNameToValueIfNeeded(path.Body.Type, value);
+            if (DocumentStore.Conventions.SaveEnumsAsIntegersForPatching && value is Enum)
+            {
+                valueToUse = Convert.ToInt32(value);
+            }
```

Comment:
this change also needs to be applied to other `Patch` overloads and `AddOrPatch`
e.g. : https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.Patch.cs#L228

take a look at these failing tests:

```
private enum Parts
{
	Engine = 42,
	Gears = 102,
	Breaks = 206
}


[RavenFact(RavenTestCategory.ClientApi)]
public void WithAddOrPatch()
{
	using var store = GetDocumentStore(new()
	{
		ModifyDocumentStore = store =>
		{
			store.Conventions.SaveEnumsAsIntegers = true;
			store.Conventions.SaveEnumsAsIntegersForPatching = true;
		},
	});

	string id;
	var entity = Machine.Create();
	using (var session = store.OpenSession())
	{
		session.Store(entity);
		session.SaveChanges();
		id = session.Advanced.GetDocumentId(entity);
	}

	using (var session = store.OpenSession())
	{
		session.Advanced.AddOrPatch(id, entity, x => x.Checks[Parts.Breaks], Status.Good);
		session.SaveChanges();
	}

	var patchedJson = GetRawJson(store, id);
	Assert.Equal("""{"Checks":{"Engine":1,"Gears":1, "Breaks":1}}""", patchedJson);  // fails, we get : {"Checks":{"Engine":1,"Gears":1,"Breaks":"Good"}}
}

[Fact]
public void WithDictionaryAdderPatch()
{
	using var store = GetDocumentStore(new()
	{
		ModifyDocumentStore = store =>
		{
			store.Conventions.SaveEnumsAsIntegers = true;
			store.Conventions.SaveEnumsAsIntegersForPatching = true;
		},
	});

	var entity = Machine.Create();
	string id;
	using (var session = store.OpenSession())
	{
		session.Store(entity);
		session.SaveChanges();
		id = session.Advanced.GetDocumentId(entity);
	}
	
	using (var session = store.OpenSession())
	{
		var item = session.Load<Machine>(id);
		session.Advanced.Patch(item, x => x.Checks, dict => dict.Add(Parts.Breaks, Status.Bad));
		session.SaveChanges();
	}
	using (var commands = store.Commands())
	{
		var item = commands.Get(id).BlittableJson;
		Assert.True(item.TryGet(nameof(Machine.Checks), out BlittableJsonReaderObject values));
		Assert.Equal(3, values.Count);

		Assert.True(values.TryGet(nameof(Parts.Engine), out int engineVal));
		Assert.Equal(1, engineVal);

		Assert.True(values.TryGet(nameof(Parts.Gears), out int gearsVal));
		Assert.Equal(1, gearsVal);

		Assert.True(values.TryGet(nameof(Parts.Breaks), out int breaksVal)); //fails, we get a string "Bad" instead of int 
		Assert.Equal(2, breaksVal);
	}
}


```

Reply:
@aviv86 Thank you for these. I have added and fixed them. I was also able to identify problems when patching lists.

I think that all cases are handled now (patch by value, with dictionary and lists).